src/main/scala/nl.knaw.dans.easy.download/EasyDownloadServlet.scala

```diff
@@ -70,6 +64,26 @@ class EasyDownloadServlet(app: EasyDownloadApp) extends ScalatraServlet
     multiParams("splat").find(!_.trim.isEmpty).map(Paths.get(_))
   }
 
+  private def download(authRequest: BasicAuthRequest, userName: String, uuid: UUID, path: Path) = {
+    getUser(authRequest, userName, uuid, path) match {
+      case Success(user) => respond(uuid, path, app.downloadFile(uuid, path, user, () => response.outputStream))
+      case Failure(InvalidUserPasswordException(_, _)) => Unauthorized()
+      case Failure(AuthenticationNotAvailableException(_)) => ServiceUnavailable("Authentication service not available, try anonymous download")
+      case Failure(AuthenticationTypeNotSupportedException(_)) => BadRequest("Only anonymous download or basic authentication supported")
+      case Failure(t) =>
+        logger.error(s"not expected exception", t)
+        InternalServerError("not expected exception")
+    }
+  }
+
+  private def getUser(authRequest: BasicAuthRequest, userName: String, uuid: UUID, path: Path): Try[Option[User]] = {
+    val fileItem = app.authorisation.getFileItem(uuid, path).toOption
```

Comment:
Please note that with this, you're doubling the workload on `easy-auth-info`! It's also called in `app.downloadFile`. Please reduce that to just one call per request to this service.

Reply:
done